products/caasp/main.pm

```diff
@@ -143,18 +144,22 @@ sub load_stack_tests {
 
 # Init cluster variables
 sub stack_init {
-    my $children      = get_children;
-    my $stack_size    = keys %$children;
-    my $stack_masters = $stack_size > 6 ? 3 : 1;             # For 6+ node clusters select 3 masters
-    my $stack_minions = $stack_size - $stack_masters - 1;    # Do not count admin node into minions
+    my $children       = get_children;
+    my $delayed_worker = get_delayed_worker;
+
+    my $stack_size    = (keys %$children) - !!$delayed_worker;
```

Comment:
is this typo or !! has some meaning in perl ?

Reply:
This translates variable into boolean value - https://stackoverflow.com/questions/2168406/what-does-double-exclamation-point-mean

Here it means - "if there is delayed worker, decrease $stack_size by 1"

Reply:
thanks for clarification